packages/grafana-ui/src/components/Monaco/types.ts

```diff
@@ -0,0 +1,77 @@
+export type CodeEditorChangeHandler = (value: string) => void;
+export type CodeEditorSuggestionProvider = () => CodeEditorSuggestionItem[];
```

Comment:
This type feels too simplistic, needs prefix & context so suggestions can be context specific. So we will have to break this api later

Reply:
For complex things, shouldn't we just use monaco/lsp?  The goal here is to have simple support for the super common templates/variables.

Complex things can be supported by either loading custom languages:
https://github.com/microsoft/monaco-languages/tree/master/src

or customizing in JS:
https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-custom-languages

Reply:
Note that these suggestions exist on top of the defined language.